activities/Vote.activity/js/components/SugarPresence.js

```diff
@@ -19,29 +19,25 @@ Vue.component('sugar-presence', {
 				// Handle shared instance
 				if (environment.sharedId) {
 					console.log("Shared instance");
-					vm.presence = activity.getPresenceObject(function (error, network) {
-						if (error) {
-							console.log(error);
-						}
-						network.onDataReceived(vm.onNetworkDataReceived);
-						network.onSharedActivityUserChanged(vm.onNetworkUserChanged);
-					});
 				}
+				vm.presence = activity.getPresenceObject(function (error, network) {
+					if (error) {
+						console.log(error);
+					}
+					network.onDataReceived(vm.onNetworkDataReceived);
+					network.onSharedActivityUserChanged(vm.onNetworkUserChanged);
+				});
 			});
 		});
 	},
 	methods: {
-
-		isNetworkAvailable() {
-			return new Promise((resolve, reject) => {
-				requirejs(["sugar-web/presence"], presence => {
-					resolve(presence.isConnected());
-				});
-			})
+		
+		isConnected: function () {
+			return this.presence != null && this.presence.isConnected();
 		},
 
-		isConnected: function () {
-			return this.presence != null;
+		isShared() {
+			return this.presence != null && this.presence.hasOwnProperty('sharedInfo');
```

Comment:
Will be better to call `this.presence.getSharedInfo()` and test if not equal to `null` instead of testing the presence of the internal property.

Reply:
Made the changes.